### PR TITLE
Refactor LicenseFragmentHelper.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Make some improvements to `LicenseFragmentHelper`.

#### Fixes the following issue(s)

- Fixes #

#### APK testing 

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.kt [hunk 1]
```diff
@@ -67,11 +67,10 @@ private fun getHexRGBColor(context: Context, color: Int): String {
...
 
 fun showLicense(context: Context?, component: SoftwareComponent): Disposable {
-    return showLicense(context, component.license) { alertDialog ->
-        alertDialog.setPositiveButton(R.string.dismiss) { dialog, _ ->
+    return showLicense(context, component.license) {
+        setPositiveButton(R.string.dismiss) { dialog, _ ->
             dialog.dismiss()
-        }
-        alertDialog.setNeutralButton(R.string.open_website_license) { _, _ ->
+        }.setNeutralButton(R.string.open_website_license) { _, _ ->
```
Comment (TacoTheDank) [inline app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.kt:66]: I would keep this on a new line, IMO.

File: app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.kt [hunk 2]
```diff
@@ -1,3 +1,5 @@
+@file:JvmName("LicenseFragmentHelper")
```
Comment (TacoTheDank) [inline app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.kt:1]: None of these are used in Java, so I don't believe we need this.

**Comment by TacoTheDank** [comment]
Also, I don't believe the default parameter commit is particularly useful.

File: app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.kt (hunk 1 above)
Comment (Isira-Seneviratne) [reply app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.kt]: > Also, I don't believe the default parameter commit is particularly useful.

It allows the method overload to be removed.

Comment (TacoTheDank) [reply app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.kt]: I'm aware, it's just that IMO it didn't really do anything useful here, considering that both `showLicense` have different block arguments for the "backend" `showLicense`'s AlertDialog. I think the previous one is simpler to read, even if it makes the class have a few more lines. If it's merged as is, then the block's default parameter (with its arguments) is just overridden by the block arguments set in `showLicense(Context, SoftwareComponent)`. I do understand the reasoning behind it, I just think that it makes it overall harder to interpret.

I can't explain very well right now, my brain is kinda fried.

**Review by Stypox — APPROVED** [review]
Tested, works, thanks. The build was failing because of ktlint, I fixed that.

**Pull request merged**